crates/matrix-sdk-ui/src/timeline/item.rs

```diff
@@ -18,6 +18,27 @@ use as_variant::as_variant;
 
 use super::{EventTimelineItem, VirtualTimelineItem};
 
+/// Opaque unique identifier for a timeline item.
+///
+/// It is transferred whenever a timeline item is updated. This can be used as a
+/// stable identifier for UI purposes, as well as operations on the event
+/// represented by the item.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Object))]
+pub struct TimelineUniqueId(pub(crate) String);
```

Comment:
I guess this is created as an `Object` instead of `Record` so it can't be instantiated from the clients? I'd rather have a plain type instead of passing a pointer as it would be better for debugging purposes though 🫤 .

Reply:
I suppose `Record` means the inner string would be available to all, and the intent here is that the identifier is an opaque thingy. Would you like a "gimme a debug string" helper for debugging purposes?

Reply:
I think that would be enough, yes. Without an event id or transaction id it could be a bit difficult to keep track of an item if it changed while debugging, with the event id back this should be better and having a way to manually check the unique id would be perfect. Thanks!